app/com/linkedin/drelephant/ElephantRunner.java

```diff
@@ -194,6 +194,10 @@ public void run() {
         if (_analyticJob != null && _analyticJob.retry()) {
           logger.error("Add analytic job id [" + _analyticJob.getAppId() + "] into the retry list.");
           _analyticJobGenerator.addIntoRetries(_analyticJob);
+        }
+        else if (_analyticJob != null && _analyticJob.secondRetry()) {
+          logger.error("Add analytic job id [" + _analyticJob.getAppId() + "] into the second retry list.");
```

Comment:
Add a comment here explaining the behavior on why we need a second retry queue.

Reply:
Done.